lib/segment/src/problems/unindexed_field.rs

```diff
@@ -0,0 +1,177 @@
+        }
+        crate::types::ValueVariants::Integer(_integer) => {
+            PayloadFieldSchema::FieldType(PayloadSchemaType::Integer)
+        }
+        crate::types::ValueVariants::Bool(_boolean) => {
+            PayloadFieldSchema::FieldType(PayloadSchemaType::Bool)
+        }
+    }
+}
+
+fn infer_type_from_any_variants(value: &AnyVariants) -> PayloadFieldSchema {
+    match value {
+        AnyVariants::Keywords(_strings) => {
+            PayloadFieldSchema::FieldType(PayloadSchemaType::Keyword)
+        }
+        AnyVariants::Integers(_integers) => {
+            PayloadFieldSchema::FieldType(PayloadSchemaType::Integer)
+        }
+    }
+}
+
+fn infer_type_from_field_condition(field_condition: &FieldCondition) -> Vec<PayloadFieldSchema> {
+    match field_condition {
+        FieldCondition {
+            r#match: Some(r#match),
+            ..
+        } => vec![match r#match {
+            Match::Value(match_value) => infer_type_from_match_value(match_value),
+            Match::Text(_match_text) => {
+                PayloadFieldSchema::FieldParams(PayloadSchemaParams::Text(TextIndexParams {
+                    r#type: TextIndexType::Text,
+                    tokenizer: TokenizerType::default(),
+                    min_token_len: None,
+                    max_token_len: None,
+                    lowercase: None,
+                }))
+            }
+            Match::Any(match_any) => infer_type_from_any_variants(&match_any.any),
+            Match::Except(match_except) => infer_type_from_any_variants(&match_except.except),
+        }],
+        FieldCondition {
+            range: Some(_range),
+            ..
+        } => vec![
+            PayloadFieldSchema::FieldType(PayloadSchemaType::Integer),
+            PayloadFieldSchema::FieldType(PayloadSchemaType::Float),
+        ],
+        FieldCondition {
+            geo_bounding_box: Some(_),
+            ..
+        }
+        | FieldCondition {
+            geo_radius: Some(_),
+            ..
+        }
+        | FieldCondition {
+            geo_polygon: Some(_),
+            ..
+        } => vec![PayloadFieldSchema::FieldType(PayloadSchemaType::Geo)],
+        FieldCondition {
+            key: _,
+            r#match: None,
+            range: None,
+            geo_bounding_box: None,
+            geo_radius: None,
+            geo_polygon: None,
+            values_count: _,
+        } => vec![],
+    }
+}
+
+impl TryFrom<(FieldCondition, String)> for UnindexedField {
+    type Error = OperationError;
+
+    /// Try to form an issue from a field condition and a collection name
+    ///
+    /// # Failures
+    ///
+    /// Will fail if the field condition cannot be used for inferring an appropriate schema.
+    /// For example, when there is no index that can be built to improve performance.
+    fn try_from(condition_collection_tuple: (FieldCondition, String)) -> Result<Self, Self::Error> {
```

Comment:
I don't think I like this too much. I find converting `(FieldCondition, String)` into an `UnindexedField` type quite arbitrary.

To use it, you'd have to construct a tuple too.

What about just:

```rust
impl UnindexedField {
    fn new(condition: FieldCondition, collection: String) { ... }
}
```